contracts/button/lib.rs

```diff
@@ -374,12 +417,8 @@ pub mod button_game {
             Ok(())
         }
 
-        fn check_role(
-            access_control: &AccessControlRef,
-            account: AccountId,
-            role: Role,
-        ) -> ButtonResult<()> {
-            if access_control.has_role(account, role) {
+        fn check_role(&self, account: AccountId, role: Role) -> ButtonResult<()> {
```

Comment:
This method is used always to check if caller is an Admin. Could you specialize it to `check_caller_is_admin` and remove args `account` and `role`?

Reply:
I will tackle this next but in another PR (https://cardinal-cryptography.atlassian.net/browse/A0-2238) and use a very similar trait abstraction for these checks once we stabilize on what syntax we want